### PR TITLE
chore: Update toolbar drawer styles

### PR DESCRIPTION
**Pull request by jkuelz**
### Description

Add toolbar visual context to apply visual styles that only apply to the toolbar (without affecting current VR or classic). For example:
- Remove the divider between panel header and content
- Reduced space between panel header and content
- Smaller button toggle size

Related links, issue #, if available: n/a

### How has this been tested?

Running through pipeline: dev-v3-jkuelz

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by just-boris** [comment]
Per our DM discussion, this change should not use visual context, but apply individual styles for each component, based on the feature flag

File: src/app-layout/visual-refresh-toolbar/toolbar/trigger-button/styles.scss [hunk 1]
```diff
@@ -61,12 +53,12 @@
...
+    'M32.185 9.34876C31.1913 10.1916 29.905 10.7 28.4999 10.7C25.3519 10.7 22.7999 8.14803 22.7999 5C22.7999 3.68192 23.2473 2.46833 23.9985 1.50281C21.8642 0.537421 19.4948 0 17 0C7.61116 0 0 7.61116 0 17C0 26.3888 7.61116 34 17 34C26.3888 34 34 26.3888 34 17C34 14.2479 33.346 11.6485 32.185 9.34876Z'
   );
 
   @include styles.with-direction('rtl') {
     clip-path: path(
-      'M1.78066 10.1845C2.6039 10.579 3.52615 10.8 4.5 10.8C7.97939 10.8 10.8 7.97939 10.8 4.5C10.8 3.52614 10.579 2.6039 10.1845 1.78066C12.5482 0.639587 15.1994 0 18 0C27.9411 0 36 8.05887 36 18C36 27.9411 27.9411 36 18 36C8.05887 36 0 27.9411 0 18C0 15.1994 0.639587 12.5482 1.78066 10.1845Z'
+      'M1.81502 9.34876C2.80869 10.1916 4.09501 10.7 5.50007 10.7C8.6481 10.7 11.2001 8.14803 11.2001 5C11.2001 3.68192 10.7527 2.46833 10.0015 1.50281C12.1358 0.537421 14.5052 0 17 0C26.3888 0 34 7.61116 34 17C34 26.3888 26.3888 34 17 34C7.61116 34 0 26.3888 0 17C0 14.2479 0.653984 11.6485 1.81502 9.34876Z'
```
Comment (just-boris) [inline src/app-layout/visual-refresh-toolbar/toolbar/trigger-button/styles.scss:61]: How is this change related to the rest of the PR?

Comment (jkuelz) [reply src/app-layout/visual-refresh-toolbar/toolbar/trigger-button/styles.scss:61]: Because the size of the trigger changed. This path is for making sure the red dot badge icon has a gap between itself and the button.

Comment (just-boris) [reply src/app-layout/visual-refresh-toolbar/toolbar/trigger-button/styles.scss:61]: Good to know, thanks. For me it sounds something to extract in a separate PR, because it may break screenshot tests (if we would have any for the toolbar mode)

Comment (jkuelz) [reply src/app-layout/visual-refresh-toolbar/toolbar/trigger-button/styles.scss:61]: These trigger styles are located in the `toolbar/` folder, so I presumed that meant they would only apply to toolbar, which is what we are wanting to change. Everything in this PR would result in a visual change for the toolbar only (on purpose). Why do you say this change be pulled out into a separate PR?

Comment (just-boris) [reply src/app-layout/visual-refresh-toolbar/toolbar/trigger-button/styles.scss:61]: Since we moved away from visual contexts, it does not matter much anymore, because it will be whole new code anyway. If you open a new PR with everything and close this one, it is also fine

**Review by just-boris — APPROVED** [review]
